pipeline/backend/kubernetes/utils.go

```diff
@@ -10,8 +12,16 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
+var dnsPattern = regexp.MustCompile(`[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`)
+
 func dnsName(i string) string {
-	return strings.Replace(i, "_", "-", -1)
+	res := strings.Replace(strings.ToLower(i), "_", "-", -1)
+
+	match := dnsPattern.MatchString(res)
+	if !match {
+		log.Panic("name not valid kubernetes DNS name")
```

Comment:
dont Panic make the dnsName func return an error to report up the chain ...

and create a new error type .. `var Name = errors.New(...)`